docs/src/internals/montecarlo.md

```diff
@@ -0,0 +1,136 @@
+    name1:name2 = 0.7
+    name1:name3 = 0.5
+
+    # assign RVs to model Parameters
+    ext_var1[2010:2049, (US, CHI)] = name1
+    ext_var1[2050:2100, (US, CHI)] = name2
+    ext_var2[2050, US] += name1
+    ext_var3 *= name3
+    ext_var4 += name1
+    ext_var5[2010:2050, :] *= name3
+    
+    # indicate which parameters to save for each model run. Specify
+    # a parameter name, or some slice of its data, similar to the
+    # assignment of RVs, above.
+    save = (param1, param2, param3[2010:2100], param4[:, (US, CHI)])
+end
+```
+
+## ModelRunner
+
+There are several types of analyses that require an ensemble of model runs performed over a set of parameter values. These include traditional Monte Carlo simulation, in which random values are drawn from distributions and applied to model parameters, as well as global sensitivity analyses that use prescribed methods for defining trial data (e.g., Sobol sampling), and Markov Chain Monte Carlo, which computes new trial data based on prior model results.
+
+The abstract type ```ModelRunner``` will be used to define a generic API for model runners, including support for parallelization of analyses on multiprocessors and cluster computing systems. The specific sampling and analysis methods required for each subtype of ```ModelRunner``` will be defined in the concrete subtype, e.g., ```MonteCarloRunner```.
+
+The generic process will look something like this:
+
+```julia
+m = ModelDef(...)
+r = McsRunner(m)
+
+# Optionally redefine random variables, overriding defaults
+setrv!(r, :rvname1, Normal(10, 3))
+
+# Optionally assign an alternative RV or distribution to a model parameter
+setparam!(r, :comp1, :param1, :rvname1)
+setparam!(r, :comp1, :param2, Uniform(0, 3))
+
+# Adjust what should be saved per model run
+@add_results(r, param10, param11[US])
+@drop_results(r, param1, param1)
+
+# Run trials 1-5000. Also can specify as a range (e.g., 5001:10000) 
+# or vector of trial numbers.
+run!(r, 5000)
+
+# Save results to the indicated directory
+write(r, dirname)
+```
+
+### Saving ensemble results
+
+Ensemble results will be stored in an object of type ```EnsembleResult```, which is used by all subtypes of ```ModelRunner```. By default, only model outputs tagged for output will be saved in the ```EnsembleResult``` instance. Parameters can be identified as "output" parameters in ```@defmcs```, and/or directly in a concrete subtype of ```ModelRunner```, in which case default values set in ```@defmcs``` can be overridden.
+
+The method ```write(r::ModelRunner, dirname::String)``` will save model results to set of CSV files in the named directory. Initially, we will store the data in [tidy](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html) format in which variables are in columns and each row represents an observation. This format is verbose but flexible and commonly used by consumers of data such as plotting packages. Other formats can be generated if the need arises.
```

Comment:
I think that makes sense, but it is not entirely clear to me how the ``EnsembleResult`` type comes into play here.